app/models/record_published_at.rb

```diff
@@ -0,0 +1,13 @@
+require "active_support/concern"
+# Asset, Work and Collection all record
+# their publication date by including this module.
+module RecordPublishedAt
+  extend ActiveSupport::Concern
+  included do
+    before_validation :set_published_at
+    private
+    def set_published_at
+      self.published_at = DateTime.now if published? && published_changed?
+    end
+  end
+end
```

Comment:
Hmm, if the tests pass this must work?  But this isn't the normal way to do it. You have to put the `before_validation` in an `included do` block... but the `def set_published_at` should not be. 

The `ActiveSupport::Concern` docs indeed aren't very clear on this. :( 

But check out [this example](https://blog.appsignal.com/2020/09/16/rails-concers-to-concern-or-not-to-concern.html).   

I think you should be able to move the `def` outside the `included do` and it will still work fine, and be a bit more standard/normal (with possibly implications for how it works). Do you mind trying it?

Except for that looks great! Go ahead and merge after that!

Reply:
Thanks again for catching that. Will merge after the tests pass.